packages/@react-spectrum/s2/src/Form.tsx

```diff
@@ -36,11 +37,29 @@ export interface FormProps extends FormStyleProps, Omit<RACFormProps, 'className
 export const FormContext = createContext<FormStyleProps | null>(null);
 export function useFormProps<T extends FormStyleProps>(props: T): T {
   let ctx = useContext(FormContext);
-  if (ctx) {
-    return {...ctx, ...props};
-  }
+  let isSkeleton = useIsSkeleton();
+  return useMemo(() => {
+    let result: T = props;
+    if (ctx || isSkeleton) {
+      result = {...props};
+    }
 
-  return props;
+    if (ctx) {
+      // This is a subset of mergeProps. We just need to merge non-undefined values.
+      for (let key in ctx) {
+        if (result[key] === undefined) {
+          result[key] = ctx[key];
+        }
+      }
+    }
+
+    // Skeleton always wins over local props.
+    if (isSkeleton) {
+      result.isDisabled = true;
+    }
+
+    return result;
+  }, [ctx, props, isSkeleton]);
```

Comment:
Is it a bad idea to do this here? It was an easy way to mark all form components within a skeleton as disabled.

Reply:
So, the alternative is to manually do this in every individual component rather than here in one place. Or I guess we could rename `useFormProps` to something else...